components/cache/cache_invalidation.rst

```diff
@@ -0,0 +1,107 @@
+
+    $item = $cache->getItem('cache_key');
+    // ...
+    // add one or more tags
+    $item->tag('tag_1');
+    $item->tag(array('tag_2', 'tag_3'));
+    $cache->save($item);
+
+If `$cache` implements :class:`Symfony\\Component\\Cache\\TagAwareAdapterInterface`,
+you can invalidate the cached items by calling
+:method:`Symfony\\Component\\Cache\\TagAwareAdapterInterface::invalidateTags`::
+
+    // invalidate all items related to `tag_1`
+    $cache->invalidateTags('tag_2');
+
+    // or invalidate all items related to `tag_1` or `tag_3`
+    $cache->invalidateTags(array('tag_1', 'tag_3'));
+
+Of course when you know the cache key, it's better to call
+``$cache->deleteItem('cache_key');``, but this key is sometimes hard to track
+and this is where cache tags become useful.
+
+Tag Aware Adapters
+~~~~~~~~~~~~~~~~~~
+
+To store tags, you need to wrap a cache adapter with the
+:class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapter` class or implement
+:class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapterInterface` and its only
+:method:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapterInterface::invalidateTags`
+method by yourself.
+
+The :class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapter` class implements
+instantaneous invalidation. It needs one or two cache adapters: the required one
+is used to store cached items; the second one is used to store tags and their
+invalidation version number (conceptually similar to their latest invalidation
+date). When only one adapter is used, items and tags are all stored in the same
+place. By using two adapters, you can e.g. store some big cached items on the
+filesystem or in the database, and keep tags in a Redis database to sync all your
+fronts and have very fast invalidation checks::
+
+    $cache = new TagAwareAdapter(
+        // Adapter for cached items
+        new FilesystemAdapter(),
+        // Adapter for tags
+        new RedisAdapter('redis://localhost')
+    );
+
+Using Cache Namespaces
+----------------------
+
+By using adapters that implement the
+:method:`Symfony\\Component\\Cache\\Adapter\\ForkableAdapterInterface::fork`
+method from :class:`Symfony\\Component\\Cache\\Adapter\\ForkableAdapterInterface`,
+you can create context-dependent variations of your cached items::
+
+    $childCache = $parentCache->fork('child-namespace-name');
+    $siblingCache = $parentCache->fork('sibling-namespace-name');
+    $subChildCache = $childCache->fork('sub-child-namespace-name');
+
+Clearing a parent adapter also clears all its forks::
+
+    $childCache->clear();
+    // both $childCache and $subChildCache are now empty
+    // but $parentCache and $siblingCache weren't affected
+
+.. note::
+
+    Invalidating by tags affects all forks.
+
+Using Cache Expiration
+----------------------
+
+If your data is valid only for a limited period of time, you can specify their
+lifetime or their expiration date with the PSR-6 interface, as explained in the
+Cache Items article.
```

Comment:
please add a reference to the cache items article: ``in the :doc:`/components/cache/cache_items` article.``